tokio/src/sync/semaphore.rs

```diff
@@ -59,18 +75,23 @@ use std::sync::Arc;
 ///     let mut join_handles = Vec::new();
 ///
 ///     for _ in 0..5 {
-///         let permit = semaphore.clone().acquire_owned().await.unwrap();
+///         let semaphore_clone = semaphore.clone();
 ///         join_handles.push(tokio::spawn(async move {
-///             // perform task...
-///             // explicitly own `permit` in the task
-///             drop(permit);
+///             perform_task(semaphore_clone).await;
 ///         }));
```

Comment:
I think we probably still want to acquire the permit before spawning. That way, we don't spawn more tasks than we have permits.

Also, with the way you're using it now, you don't need `acquire_owned`. Using `acquire` would work fine, since the permit itself is not moved across tasks.

Reply:
Fair enough. I was assuming you would queue up the tasks, and they would run non-stop as permits were free'd -- but this makes more sense resource-wise. I'll change it back to the original, and rework the doc string.

Can you tell me the significance about moving permits (or anything for that matter) across tasks? Is this a matter of scoping and lifetimes?

Reply:
Ahh, I understand. Thank you.

How's this?